sda/processor.go

```diff
@@ -70,10 +74,13 @@ func (d *BlockingDispatcher) RegisterProcessor(p Processor, msgType ...network.M
 // Dispatch will directly call the right processor's method Process. It's a
 // blocking call if the Processor is blocking !
 func (d *BlockingDispatcher) Dispatch(packet *network.Packet) error {
+	d.Lock()
 	var p Processor
 	if p = d.procs[packet.MsgType]; p == nil {
-		return errors.New("No Processor attached to this message type.")
+		d.Unlock()
+		return errors.New("No Processor attached to this message type " + packet.MsgType.String())
 	}
+	d.Unlock()
```

Comment:
I wanted to unlock before the processor gets the message otherwise no other processor could be called at the same time (by different go routines for example). Only one processor could be having a message at a time.